common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesAuthorityValidationMetadataCache.java

```diff
@@ -0,0 +1,84 @@
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.cache;
+
+import android.content.Context;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import com.microsoft.identity.common.adal.internal.cache.IStorageHelper;
+import com.microsoft.identity.common.adal.internal.cache.StorageHelper;
+
+public class SharedPreferencesAuthorityValidationMetadataCache implements IDefaultAuthorityValidationMetadataCache {
+
+    private static final String CACHE_VALUE_SEPARATOR = "-";
+    private static final String AUTHORITY_VALIDATION_METADATA_CACHE_GUID = "33DD5583-1098-4617-AF07-2D327BC4C0E4";
+
+    /**
+     * The name of the SharedPreferences file on disk.
+     * This file is only used to store MSAL CPP authority validation metadata
+     */
+    private static final String DEFAULT_CPP_AUTHORITY_VALIDATION_METADATA_SHARED_PREFERENCES =
+            "com.microsoft.identity.client.cpp_authority_validation_metadata";
+
+    private final ICacheKeyValueDelegate mCacheKeyValueDelegate;
+
+    private final ISharedPreferencesFileManager mCppAuthorityValidationMetadataSharedPreferencesFileManager;
+
+    public SharedPreferencesAuthorityValidationMetadataCache(final Context context) {
+        mCacheKeyValueDelegate = new CacheKeyValueDelegate();
+
+        // We don't need to encrypt this cache file, since the cache file does not contain any
+        // secrets and other apps do not have access to this cache file.
+        final IStorageHelper storageHelper = null;
+        mCppAuthorityValidationMetadataSharedPreferencesFileManager =
+                SharedPreferencesFileManager.getSharedPreferences(
+                        context,
+                        DEFAULT_CPP_AUTHORITY_VALIDATION_METADATA_SHARED_PREFERENCES,
+                        storageHelper
+                );
+    }
+
+    @Override
+    public String generateAuthorityValidationMetadataKey(final String environment) {
+        return AUTHORITY_VALIDATION_METADATA_CACHE_GUID + CACHE_VALUE_SEPARATOR + environment;
+    }
+
+    @Override
+    public void saveAuthorityValidationMetadata(final String environment, final String cacheValue) {

```

Comment:
Is `null` allowed? Might want to address this in javadoc

Reply:
I added `@NonNull` in the parameter of the function, and throw ClientException if null in the implementation. Is this the right  approach?